crates/bevy_ui/src/geometry.rs

```diff
@@ -66,61 +67,60 @@ use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign};
 /// right values of the position because the size of the UI element is already explicitly specified.
 ///
 /// ```
-/// # use bevy_ui::{UiRect, Size, Val, Style};
+/// # use bevy_ui::{UiRect, Size, Val, SizeConstraints, Offset};
 /// # use bevy_utils::default;
 /// #
-/// let style = Style {
-///     position: UiRect { // Defining all four sides
-///         left: Val::Px(100.0),
-///         right: Val::Px(200.0),
-///         top: Val::Px(300.0),
-///         bottom: Val::Px(400.0),
-///     },
-///     size: Size::new(Val::Percent(100.0), Val::Percent(50.0)), // but also explicitly specifying a size
-///     ..default()
-/// };
+/// let position_offset = Offset(UiRect {
+///     left: Val::Px(100.0),
+///     right: Val::Px(200.0),
+///     top: Val::Px(300.0),
+///     bottom: Val::Px(400.0),
+/// });
+/// /// Specify a size
+/// let size_constraints = SizeConstraints::suggested(Val::Percent(100.0), Val::Percent(50.0));
+///
 /// ```
 ///
 /// ## Margin
 ///
 /// A margin is used to create space around UI elements, outside of any defined borders.
 ///
 /// ```
-/// # use bevy_ui::{UiRect, Val};
+/// # use bevy_ui::{UiRect, Val, Spacing};
 /// #
-/// let margin = UiRect::all(Val::Auto); // Centers the UI element
+/// let margin = Spacing::margin(UiRect::all(Val::Auto)); // Centers the UI element
 /// ```
 ///
 /// ## Padding
 ///
 /// A padding is used to create space around UI elements, inside of any defined borders.
 ///
 /// ```
-/// # use bevy_ui::{UiRect, Val};
+/// # use bevy_ui::{UiRect, Val, Spacing};
 /// #
-/// let padding = UiRect {
+/// let padding = Spacing::padding(UiRect {
 ///     left: Val::Px(10.0),
 ///     right: Val::Px(20.0),
 ///     top: Val::Px(30.0),
 ///     bottom: Val::Px(40.0),
-/// };
+/// });
 /// ```
 ///
 /// ## Borders
 ///
 /// A border is used to define the width of the border of a UI element.
 ///
 /// ```
-/// # use bevy_ui::{UiRect, Val};
+/// # use bevy_ui::{UiRect, Val, Spacing};
 /// #
-/// let border = UiRect {
+/// let border = Spacing::border(UiRect {
 ///     left: Val::Px(10.0),
 ///     right: Val::Px(20.0),
 ///     top: Val::Px(30.0),
 ///     bottom: Val::Px(40.0),
-/// };
+/// });
 /// ```
-#[derive(Copy, Clone, PartialEq, Debug, Default, Reflect)]
+#[derive(Copy, Clone, PartialEq, Debug, Default, Reflect, Serialize, Deserialize)]
```

Comment:
Reviewers: Now that `Style` is no more and used slightly differently, please do a documentation pass to check if this particular explanation and examples still makes the most sense.

Reply:
I'm relatively happy with this, but I think that the margin/padding/border explanation would make more sense on the `Spacing` struct.